Tasks/JenkinsDownloadArtifacts/ArtifactDetails/Downloader.ts

```diff
@@ -0,0 +1,125 @@
+import * as Q from 'q';
+import * as  tl from 'vsts-task-lib/task';
+
+import {JenkinsRestClient} from "./JenkinsRestClient"
+import {CommitsDownloader} from "./CommitsDownloader"
+import {WorkItemsDownloader} from "./WorkItemsDownloader"
+
+var handlebars = require('handlebars');
+
+export class ArtifactDetailsDownloader {
+    public DownloadCommitsAndWorkItems(): Q.Promise<void> {
+        let defer: Q.Deferred<any> = Q.defer<any>();
+
+        console.log(tl.loc("DownloadingCommitsAndWorkItems"));
+        
+        let startJobId: number = parseInt(tl.getInput("previousJenkinsBuildNumber"))
+        let commitsDownloader: CommitsDownloader = new CommitsDownloader();
+
+        if (!startJobId) {
+            let jenkinsBuild = tl.getInput('jenkinsBuild', true);
+            let endJob: string = "";
+
+            if (jenkinsBuild === 'LastSuccessfulBuild') {
+                // If its LastSuccessfulBuild, we don't need to fetch the build number, we could just use the lastSuccessfulFull macro
+                endJob = 'lastSuccessfulBuild';
+            }
+            else {
+                // if its not LastSuccessfulBuild try to get the build number from input.
+                endJob = tl.getInput("jenkinsBuildNumber", false)
+            }
+
+            console.log(tl.loc("JenkinsDownloadingChangeFromCurrentBuild", endJob));
+
+            commitsDownloader.DownloadFromSingleBuildAndSave(endJob).then((commits: string) => {
+                    let commitMessages: string[] = CommitsDownloader.GetCommitMessagesFromCommits(commits);
+                    let workItemsDownloader: WorkItemsDownloader = new WorkItemsDownloader(commitMessages); 
+                    workItemsDownloader.DownloadFromSingleBuildAndSave(endJob).then(() => {
+                        defer.resolve(null);
+                    }, (error) => {
+                        defer.reject(error);
+                    });
+            }, (error) => {
+                defer.reject(error);
+            });
+        }
+        else {
+            let endJobId = parseInt(tl.getInput("jenkinsBuildNumber", true));
+            if (startJobId < endJobId) {
+                console.log(tl.loc("DownloadingJenkinsChangeBetween", startJobId, endJobId));
+            }
+            else if (startJobId > endJobId) {
+                console.log(tl.loc("JenkinsRollbackDeployment", startJobId, endJobId));
```

Comment:
didn't see this resource being defined

you can word it as:

Handling rollback deployment, start build (%s) is greater than end build (%s)

Reply:
done